linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -153,14 +149,41 @@ final class Emitter private (config: CommonPhaseConfig,
       val trees = mutable.ListBuffer.empty[js.Tree]
 
       logger.time("Emitter: Write trees") {
-        val WithGlobals(coreJSLibTree, coreJSLibTrackedGlobalRefs) =
-          state.coreJSLibCache.tree
+        val WithGlobals(coreJSLib, coreJSLibTrackedGlobalRefs) =
+          state.coreJSLibCache.lib
 
-        trees += coreJSLibTree
+        trees += coreJSLib.definitions
 
         emitModuleImports(orderedClasses, trees, logger)
 
-        emitGeneratedClasses(trees, generatedClasses)
+        /* Emit all the classes, in the appropriate order:
```

Comment:
If you inline the contents of `emitGeneratedClasses`, perhaps it would be good to move this big comment at the beginning (before `trees += coreJSLib.definitions`), and explicitly list the steps of 1) emit the definitions of the core JS lib and 2) emit the module imports.

That would also be nicer in the next commit, as then the comment would apply to the entire iterator concatenation, rather than being in the middle of it.

Reply:
Good point. I have even split the comment over the multiple lines. Like this, it is much less likely to get out of sync.